libvroom_examples/libvroom.cpp

```diff
@@ -97,7 +97,7 @@ void run_example_with_osrm() {
 
   // Create one-dimension capacity restrictions to model the situation
   // where one vehicle can handle 4 jobs with deliveries.
-  problem_instance.set_amount_size(amount_dimension);
+  problem_instance.set_zero_amount(vroom::Amount(amount_dimension));
```

Comment:
I see several drawbacks for changing from `Input::set_amount_size` to `Input::set_zero_amount`:
- in this example, client code is more complex, and in not way "oblivious to the fact that the amount has some dimensions";
- instead of just dealing with dimension (which is something the client code should know as part of the model) this requires to know how to build a zero `Amount` from its dimension which we could consider internals;
- this is dangerous since then any caller could initiate the zero member in `Input` with any non-zero amount.